### ISSUE TITLE
Steam not correctly reporting uninstalled games

### ISSUE THREAD

**Issue by sengtsan** [JosefNemec/Playnite#2043]
**Describe the bug**

- When updating the library, Steam games I have uninstalled are no longer being reported as uninstalled. I have been trialling a number of demos which are being added when installed, but not reported as uninstalled when this has happened.  

**To Reproduce**
The following titles which i have uninstalled which Playnite is not recognising:

Fight in Tight Spaces
Orange Cast
Papetura
Xuan Yuan Sword VII

**Diagnostics ID**
55ce3401-bbed-4240-8950-ef81e06ebf5a

**Comment by JosefNemec** [comment]
I don't understand, are you saying that those game are not at all visible in Playnite (but used to be) or that they are being reported as installed in Playnite (but should not be installed)?

**Comment by sengtsan** [comment]
The latter, 
The procedure is as follows:
I installed the games on Steam. 
Then I updated Playnite and the games appeared as expected. 
Then I went back to Steam and uninstalled the games. 
Then I went back to Playnite and updated the library, but those games still appeared as installed.

**Comment by darklinkpower** [comment]
Probably damaged installation. Check https://github.com/JosefNemec/Playnite/wiki/Steam-troubleshooting#installed-games-are-not-detected

**Comment by JosefNemec** [comment]
Do you have import of uninstalled games enabled? Installation state won't be set to uninstalled unless that option is enabled.

**Comment by sengtsan** [comment]
I have uninstalled then reinstalled Steam but it hasn't resolved the issue. I can confirm the import of uninstalled games setting is enabled.

![image](https://user-images.githubusercontent.com/63714826/95650732-4fee4e80-0add-11eb-8546-1161d7dcc751.png)

all the settings necessary are public on Steam

![image](https://user-images.githubusercontent.com/63714826/95650873-62b55300-0ade-11eb-9bc6-03cf39c86086.png)

**Comment by JosefNemec** [comment]
No idea what's happening then. I don't see any errors in logs and I can't reproduce it myself. If you edit the install state manually, does it stay that way after updating library?

**Comment by sengtsan** [comment]
VERY strange. 

1. I marked  a game demo that i have recently installed and then uninstalled as "installed" manually. When I do a Playnite library refresh it doesn't notice that is uninstalled and doesn't change the game status to uninstalled. (example used is Manifold Garden demo)

2.  I marked a game demo I recently installed as uninstalled manually. When I do a playnite library refresh it doesn't notice that the demo is installed and the game still is marked as uninstalled in Playnite. (example used is Orange Cast demo)

This behaviour is not seen for all demos however. I installed the Relicta demo, and that is properly updated in playnite when the library is refreshed. 

The strange behaviour is also not seen for full games in my library.

3. I uninstalled "Flower" then updated playnite. It recognised the uninstall. 
I then reinstalled "Flower" then updated playnite. It recognised the install. 

4. I marked a previously installed game, Life is Strange 2, as uninstalled in Playnite and then refreshed the library. As expected the entry updated to show as installed.

In summary, it looks like there is an issue with SOME of the recent demos. I experienced issues with the following demos:

Fight in Tight Spaces
Hellish Quart
Manifold Garden
Orange Cast
Papetura
Xuan Yuan Sword VII

but not with these demos:
Relicta
Lucifer Within Us
F I S T Forged in Shadow Torch
Paradise Lost 

OR with any full titles already in my collection. 

A bit confusing. 










  

With these latest Steam demos during this Autumn promotion, it is behaving as per my earlier message.

**Comment by sengtsan** [comment]
Something else. 
I installed free to play game Paunch and then uninstalled it. Playnite does not recognise the uninstall after refreshing the playnite library. I tried deleting the common apps folder for the game and retrying the library refresh but Playnite still saw the game as installed. 

I then removed the game from the Playnite library completely and then ran the library refresh. I found the game is no longer recognised in Playnite. N.B. the game does still appear in my Steam library even though it is free to play.

**Comment by JosefNemec** [comment]
Thanks for the additional info. If it's only an issue with some games, then it's the same problem as #910 and will be fixed by #1808. Steam API doesn't report some games/demos as part of your library, which prevents Playnite from properly setting the install status.

**Comment by sengtsan** [comment]
Ok, thanks for looking at this. All the best.

**Comment by JosefNemec** [comment]
I'm closing this as duplicate to #1808, since it should be fixed by #1808 in Playnite 9. If it's still an issue then, please open new issue, thanks.
